charmcraft.yaml

```diff
@@ -5,9 +5,19 @@ type: "charm"
 # in focal doesn't work in jammy and vice versa.
 parts:
   charm:
-    build-packages: [git]
-    charm-python-packages: [setuptools < 58]
-    prime: ["scripts"]
+    plugin: charm
+    source: .
+    charm-binary-python-packages:
+      # use a binary distribution rather than build from public source code,
+      # and it reduces a bunch of unnecessary build dependencies
+      - juju
+    build-packages:
+      - git
+  scripts:
+    plugin: dump
+    source: .
+    prime:
+      - scripts
```

Comment:
Why do we need these extra structure changes?  Can we not simply add juju to python binary packages without changing anything else?

Reply:
thanks, that's correct, this also need reverting (not really needed if we pinned)